### PR TITLE
Add Pre-commit, flake8 conf, gitignore etc to Upgrade

### PR DESCRIPTION
**Pull request by jyejare**
This PR includes:

A Linting improvements by adding :
--------

- [x] Pre-commit hooks

.flake8 Conf:
-------

- [x] flake8 conf to ingnore some errors and to specific files

.gitignore to ignore some files from gits eye:
-------

- [x] some log files, packaging folders and pycharm

Adding pre-commit in optional requirements:
-------

- [x] `pre-commit` package added

And many Files changed due to pre-commit hook auto updates.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+#     echo *.swp >> ~/.gitignore_global
+#
+
+# Compiled python code.
+*.pyc
+
+# Upgrade log files
+upgrade_highlights
+full_upgrade
+
```
Comment (devendra104) [inline .gitignore:16]: Please add "preupgrade_cli", "preupgrade_api", "postupgrade_cli", "postupgrade_api"  in ignore list.

File: .gitignore [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+# Product Setup for upgrade scenarios
+product_setup
+
+# Distribution / packaging
+satellite6_upgrade.egg-info/
+.eggs/
+
```
Comment (devendra104) [inline .gitignore:23]: add  `__pycache__` check.

**Comment by jyejare** [comment]
@devendra104 Fixed your comments. Revisit plz.

**Comment by JacobCallahan** [comment]
@jyejare has this been tested?

**Comment by jyejare** [comment]
Yes and no need :) Travis is green.

**Comment by jyejare** [comment]
@JacobCallahan Yes and no need, Travis is green :)

**Pull request merged**